modules/core/aws-batch/stack.py

```diff
@@ -0,0 +1,242 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License").
+#    You may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import logging
+from typing import Any, Dict, cast
+
+import aws_cdk.aws_batch_alpha as batch
+import aws_cdk.aws_ec2 as ec2
+import aws_cdk.aws_iam as iam
+import cdk_nag
+from aws_cdk import Aspects, Stack, Tags
+from cdk_nag import NagSuppressions
+from constructs import Construct, IConstruct
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+DEFAULT_MAX_VCPUS_PER_QUEUE = str(256)
+
+
+class AwsBatch(Stack):  # type: ignore
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        *,
+        config: Dict[str, Any],
```

Comment:
We have found that declaratively passing in the parameters as opposed to a dict makes it much easier to debug and manage.  But, lets discuss if that is best going forward?

Reply:
Sounds good, I actually had originally implemented it like that and then changed it to follow the EKS module which had the config dict. Will refactor